docs/concepts-and-explanations/javatemplate.md

```diff
@@ -274,4 +275,39 @@ When a JavaTemplate is used for matching, then the `Matcher` can be used to "ext
 
 This can then be passed into another JavaTemplate to generate some new code (such as what was done with the `after` template above).
 
-Another useful example that doesn't involve Refaster recipes is our [JavaTemplateMatchTest class](https://github.com/openrewrite/rewrite/blob/main/rewrite-java-test/src/test/java/org/openrewrite/java/JavaTemplateMatchTest.java#L32). In there, you can see how we create a JavaTemplate for the use case of matching/finding code rather than replacing it.
\ No newline at end of file
+Another useful example that doesn't involve Refaster recipes is our [JavaTemplateMatchTest class](https://github.com/openrewrite/rewrite/blob/main/rewrite-java-test/src/test/java/org/openrewrite/java/JavaTemplateMatchTest.java#L32). In there, you can see how we create a JavaTemplate for the use case of matching/finding code rather than replacing it.
+
+## Semantics class shortcut
+
+If you're looking for a more concise way to define JavaTemplates, the [Semantics class](https://github.com/openrewrite/rewrite-templating/blob/main/src/main/java/org/openrewrite/java/template/Semantics.java) offers a powerful alternative. Instead of writing the raw template string yourself, you can define the desired expression or statement directly in Java, and a annotation processor will automatically generate the corresponding template code for you.
+
+This approach helps you avoid dealing with the syntax of raw templates and lets the compiler validate your logic directly.
+
+For example, instead of writing:
+
+```java
+JavaTemplate isEmptyReplacement =
+        JavaTemplate.builder("(#{any(java.lang.String)} == null || #{any(java.lang.String)}.isEmpty())").build();
+```
+
+You can write:
+
+```java
+JavaTemplate isEmptyReplacement =
+        Semantics.expression(this, "IsEmpty", (String s) -> (s == null || s.isEmpty())).build();
+```
+
+Similarly, for statements:
+
+```java
+JavaTemplate newInstanceTemplate =
+        Semantics.statement(this, "Example", () -> { String example = "some statement"; }).build();
+```
+
+Behind the scenes, the annotation processor transforms these `Semantics.expression(...)` and `Semantics.statement(...)` calls into actual JavaTemplate definitions with proper substitution indicators and imports.
+
+:::info
+Prerequisite: To use this feature, make sure your project is set up for Refaster-style recipes with annotation processing enabled. See the [How to create a Refaster recipe](../authoring-recipes/refaster-recipes#how-to-create-a-refaster-recipe) section for more details.
+:::
+
+This pattern is especially useful when you're prototyping or when your templates are derived from simple expressions or statements that can be represented as lambdas. It's functionally equivalent to defining a JavaTemplate manually but with less boilerplate.
```

Comment:
I only ever recall seeing `Semantics` used to match constructs where it would be awkward to match the individual LST elements, but I don't see that reflected in the docs here. I'd think we want to at least make that intended usage clear.

Not sure if `Semantics` has been tried as the source of a JavaTemplate that you _apply_, as opposed to _match_. 🤔 For the moment I have a bit of trouble thinking of a direct use case where Refaster would not be better used. Perhaps another reason to document the intended use.